### ISSUE TITLE
Pass cache result into ttl function

### ISSUE THREAD

**Issue by Rebzzel** [Krukov/cashews#149]
With new exceptions api (#143), this feature may be super useful.

```py
def my_ttl_calculation(a: int, prev_cache_result: Any | None):
    if isinstance(prev_cache_result, MyException):
        return "5m"
    return "10s"
    

@cache(ttl=my_ttl_calculation)    
def my_func(a: int):
    ...
```

**Comment by Krukov** [comment]
Thanks , I'll think about this feature 

Am I right that prev_cache_result is a last result in cache ?   in this case it is only possible  with  early ( https://github.com/Krukov/cashews#early ) or hit strategy ( cached value still in cache)

**Comment by Rebzzel** [comment]
Yes, that's right. But I have another better idea - we can calculate ttl after receiving result. Then this becomes really powerful.

**Comment by Rebzzel** [comment]
So this becomes:
```py
def my_ttl_calculation(a: int, result: Any):
    if isinstance(result, MyException):
        return "5m"
    return "10s"
    

@cache(ttl=my_ttl_calculation)    
def my_func(a: int):
    ...
```

**Comment by Krukov** [comment]
yep I like it more then passing prev result.  I'll definitely implement it

**Comment by Krukov** [comment]
released 6.3.0
